src/Microsoft.DotNet.Cli.Utils/CommandResolution/DepsJsonCommandResolver.cs

```diff
@@ -0,0 +1,261 @@
+        {
+            var commandCandidates = new List<CommandCandidate>();
+
+            var assemblyCommandCandidates = GetAssemblyCommandCandidates(commandName, dependencyContext);
+            var nativeCommandCandidates = GetNativeCommandCandidates(commandName, dependencyContext);
+
+            commandCandidates.AddRange(assemblyCommandCandidates);
+            commandCandidates.AddRange(nativeCommandCandidates);
+
+            var command = ChooseCommandCandidate(commandCandidates);
+
+            return command?.GetAbsoluteCommandPath(_nugetPackageRoot);
+        }
+
+        private IEnumerable<CommandCandidate> GetAssemblyCommandCandidates(string commandName, DependencyContext dependencyContext)
+        {
+            var commandCandidates = new List<CommandCandidate>();
+
+            foreach (var runtimeLibrary in dependencyContext.RuntimeLibraries)
+            {
+                var runtimeAssetGroups = runtimeLibrary.RuntimeAssemblyGroups;
+
+                commandCandidates.AddRange(GetCommandCandidatesFromRuntimeAssetGroups(
+                    commandName, 
+                    runtimeAssetGroups,
+                    runtimeLibrary.Name,
+                    runtimeLibrary.Version));
+            }
+
+            return commandCandidates;
+        }
+
+        private IEnumerable<CommandCandidate> GetNativeCommandCandidates(string commandName, DependencyContext dependencyContext)
+        {
+            var commandCandidates = new List<CommandCandidate>();
+
+            foreach (var runtimeLibrary in dependencyContext.RuntimeLibraries)
+            {
+                var runtimeAssetGroups = runtimeLibrary.NativeLibraryGroups;
+
+                commandCandidates.AddRange(GetCommandCandidatesFromRuntimeAssetGroups(
+                                    commandName,
+                                    runtimeAssetGroups,
+                                    runtimeLibrary.Name,
+                                    runtimeLibrary.Version));            
+            }
+
+            return commandCandidates;
+        }
+
+        private IEnumerable<CommandCandidate> GetCommandCandidatesFromRuntimeAssetGroups(
+            string commandName,
+            IEnumerable<RuntimeAssetGroup> runtimeAssetGroups, 
+            string PackageName,
+            string PackageVersion)
+        {
+            var candidateAssetGroups = runtimeAssetGroups
+                .Where(r => r.Runtime == string.Empty)
+                .Where(a => 
+                    a.AssetPaths.Any(p => Path.GetFileNameWithoutExtension(p)
+                    .Equals(commandName, StringComparison.OrdinalIgnoreCase)));
+
+            var commandCandidates = new List<CommandCandidate>();
+            foreach (var candidateAssetGroup in candidateAssetGroups)
+            {
+                var candidateAssetPaths = candidateAssetGroup.AssetPaths.Where(
```

Comment:
Can you just do Select(s => new ...) after the where? If it is not that easy, forget out it.